MacOS-10/BlueBubblesHelper/BlueBubblesHelper.m

```diff
@@ -558,6 +615,129 @@ +(IMFileTransfer *) prepareFileTransferForAttachment:(NSURL *) originalPath file
     return newTransfer;
 }
 
+/**
+ Starts the listener for alias status changes
+ */
++(void) registerListenerForActiveAliasChanged {
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(_aliasesChanged:) name:@"IMAccountAliasesChangedNotification" object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(_aliasesChanged:) name:@"SOAccountAliasesChangedNotification_Private" object:nil];
+}
+/**
+ Internal reaction to notifications about aliases
+ @param notification the object inside of the notification will always be a SOAccountAliasController
+ */
++(void)_aliasesChanged:  (NSNotification *)notification{
+    SOAccountAliasController * account = notification.object;
+
+    NSArray* currentAliases = [BlueBubblesHelper getVettedAliases];
+    DLog(@"BLUEBUBBLESHELPERF: Aliases Changed %@", currentAliases);
+    [[NetworkController sharedInstance] sendMessage: @{@"event": @"aliases-updated", @"aliases": currentAliases}];
```

Comment:
Rather than sending the current aliases is there any way we can get the delta (ie which alias deactivated / activated)? I think that would be more useful as a notification rather than just listing the currently active aliases.

Reply:
That would probably easier parsed from the server side. Sometimes the notification can be triggered by just restarting imessage.app @tneotia

Reply:
Maybe `userInfo` has data? Probably not because private notification but worth checking.

Reply:
Nothing in the `userInfo`